tests/ImageSharp.Tests/Drawing/Text/DrawTextOnImageTests.cs

```diff
@@ -101,8 +107,151 @@ public class DrawTextOnImageTests
+
+        [Theory]
+        [WithSolidFilledImages(200, 100, "White", PixelTypes.Rgba32, 50, "SixLaborsSampleAB.woff", AB)]
+        [WithSolidFilledImages(900, 100, "White", PixelTypes.Rgba32, 50, "OpenSans-Regular.ttf", TestText)]
+        public void FontShapesAreRenderedCorrectlyAlongAPath<TPixel>(
+            TestImageProvider<TPixel> provider,
+            int fontSize,
+            string fontName,
+            string text)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Font font = CreateFont(fontName, fontSize);
+            TPixel colorFill = NamedColors<TPixel>.Gray;
+            TPixel colorOutline = NamedColors<TPixel>.Black;
+            IBrush<TPixel> fillBrush = Brushes.Solid(colorFill);
+            IPen<TPixel> outlinePen = Pens.DashDot(colorOutline, 3);
+
+            provider.VerifyOperation(
+                TextDrawingComparer,
+                img =>
+                    {
+                        IPath path = new Path(new LinearLineSegment(new Point(0, img.Height), new Point(img.Width, 0)));
+                        img.Mutate(
+                            c =>
+                                {
+                                    c.DrawText(
+                                        new TextGraphicsOptions
+                                            {
+                                                HorizontalAlignment = HorizontalAlignment.Center,
+                                                VerticalAlignment = VerticalAlignment.Top
+                                            },
+                                        text,
+                                        new Font(font, fontSize),
+                                        fillBrush,
+                                        outlinePen,
+                                        path);
+                                });
+                    },
+                $"pen_{fontName}-{fontSize}-{ToTestOutputDisplayText(text)}",
+                appendPixelTypeToFileName: false,
+                appendSourceFileOrDescription: true);
+        }
+
+        [Theory]
+        [WithSolidFilledImages(600, 600, "White", PixelTypes.Rgba32, 50, "OpenSans-Regular.ttf", TestText)]
+        public void FontShapesAreRenderedCorrectlyAlongACirclePath<TPixel>(
+            TestImageProvider<TPixel> provider,
+            int fontSize,
+            string fontName,
+            string text)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            Font font = CreateFont(fontName, fontSize);
+            TPixel colorFill = NamedColors<TPixel>.Black;
+            IBrush<TPixel> fillBrush = Brushes.Solid(colorFill);
+            
+            provider.VerifyOperation(
+                TextDrawingComparer,
+                img =>
+                {
+                    int w = (int)(img.Width * 0.6);
+                    int h = (int)(img.Height * 0.6);
+                    IPath path = new EllipsePolygon(img.Width/2, img.Height/2, w, h);
```

Comment:
@tocsoft can you confirm, if the output is correct?

![image](https://user-images.githubusercontent.com/6835152/41508910-054fcfba-724c-11e8-96d8-68be29e48468.png)

The first point in `path` is `X=120, Y=300`, but the text starts on the top, so I'm a bit confused. Is it center aligned?

Reply:
I've been thinking about this and think we should just remove this set of APIs... its just a thin wrapper around `TextBuilder.GenerateGlyphs` from Shapes and you can already draw shapes... it would also encourage users to investigate the shapes apis and also allow us to drop the SixLabors.Shapes.Text package and switch down to just Fonts & Shapes directly, then is users want to do this we can direct them to the optional package that they can use.

Reply:
it doesn't harm that its one less API to make sure is displaying correctly before we launch too.

Reply:
The TextBuilder.GenerateGlyphs are really tests that belong in the Shapes project and not really Imagesharp concerns... as far and Imagesharp is concerned they will be paths and that it renders paths correctly.